File: complaint-system-frontend/src/app/services/complaint.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Complaint } from '../model/complaint.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ComplaintService {

  baseUrl:string = "http://localhost:8080/api/complaint"
  constructor(private http:HttpClient) { }

  addComplaint(complaint:Complaint){
    return this.http.post(this.baseUrl,complaint);
  }

  getAllComplaints(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl);
  }

  getRaisedComplaints(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/raised");
  }

  getAllComplaintsManagerA(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/getAllCompManagerA");
  }

  getAllComplaintsManagerB(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/getAllCompManagerB");
  }

  getAllComplaintsManagerC(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/getAllCompManagerC");
  }

  getAllComplaintsManagerD(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/getAllCompManagerD");
  }

  getAllComplaintsManagerE(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/getAllCompManagerE");
  }

  getAllComplaintsManagerF(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/getAllCompManagerF");
  }

  getAllComplaintsManagerG(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/getAllCompManagerG");
  }

  getAllComplaintsManagerH(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/getAllCompManagerH");
  }

  getAllComplaintsManagerI(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/getAllCompManagerI");
  }

  getAllComplaintsManagerJ(): Observable<Complaint[]>{
    return this.http.get<Complaint[]>(this.baseUrl+"/getAllCompManagerJ");
  }
}

File: Complaint-Redressal-System/src/main/java/com/complaint/system/service/ComplaintService.java
package com.complaint.system.service;

import java.util.ArrayList;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.complaint.system.entity.Complaint;
import com.complaint.system.repository.ComplaintRepo;

@Service
public class ComplaintService {

	@Autowired
	ComplaintRepo complaintRepo;
	
	public Complaint addComplaint(Complaint complaint) {
		return complaintRepo.save(complaint);
	}
	
	public List<Complaint> getAllComplaints(){
		return complaintRepo.findAll();
	}

	public void delComplaint(int id) {
		complaintRepo.deleteById(id);
	}
	
	public List<Complaint> getAllRaisedComplaints(){
		List<Complaint> allComp = complaintRepo.findAll();
		List<Complaint> raiseComp = new ArrayList<Complaint>();
		for (Complaint complaint : allComp) {
			if(complaint.isStatus()!=true) {
				raiseComp.add(complaint);
			}
		}
		return raiseComp;
	}
	
	public List<Complaint> getAllCompManagerA(){
		return complaintRepo.getAllComplaintsManagerA();
	}
	
	public List<Complaint> getAllCompManagerB(){
		return complaintRepo.getAllComplaintsManagerB();
	}
	
	public List<Complaint> getAllCompManagerC(){
		return complaintRepo.getAllComplaintsManagerC();
	}
	
	public List<Complaint> getAllCompManagerD(){
		return complaintRepo.getAllComplaintsManagerD();
	}
	
	public List<Complaint> getAllCompManagerE(){
		return complaintRepo.getAllComplaintsManagerE();
	}
	
	public List<Complaint> getAllCompManagerF(){
		return complaintRepo.getAllComplaintsManagerF();
	}
	
	public List<Complaint> getAllCompManagerG(){
		return complaintRepo.getAllComplaintsManagerG();
	}
	
	public List<Complaint> getAllCompManagerH(){
		return complaintRepo.getAllComplaintsManagerH();
	}
	
	public List<Complaint> getAllCompManagerI(){
		return complaintRepo.getAllComplaintsManagerI();
	}
	
	public List<Complaint> getAllCompManagerJ(){
		return complaintRepo.getAllComplaintsManagerJ();
	}
}
